src/Zio/UPathExtensions.cs

```diff
@@ -239,28 +239,26 @@ public static bool IsInDirectory(this UPath path, UPath directory, bool recursiv
     /// </summary>
     /// <param name="path">The path.</param>
     /// <param name="name">The name of a parameter to include n the <see cref="ArgumentNullException"/>.</param>
-    /// <returns>A path not modified.</returns>
     /// <exception cref="System.ArgumentNullException">If the path was null using the parameter name from <paramref name="name"/></exception>
-    public static UPath AssertNotNull(this UPath path, string name = "path")
+    public static void AssertNotNull(this UPath path, string name = "path")
     {
-        if (path.FullName is null)
-            throw new ArgumentNullException(name);
-        return path;
+        if (path.IsNull)
+            Throw(name);
+
+        static void Throw(string name) => throw new ArgumentNullException(name);
     }
 
     /// <summary>
     /// Asserts the specified path is absolute.
     /// </summary>
     /// <param name="path">The path.</param>
     /// <param name="name">The name of a parameter to include n the <see cref="ArgumentNullException"/>.</param>
-    /// <returns>A path not modified.</returns>
     /// <exception cref="System.ArgumentException">If the path is not absolute using the parameter name from <paramref name="name"/></exception>
-    public static UPath AssertAbsolute(this UPath path, string name = "path")
+    public static void AssertAbsolute(this UPath path, string name = "path")
     {
-        AssertNotNull(path, name);
```

Comment:
I removed the explicit null assert, as if it is null, it can never be absolute. This is the cause of the failing tests, let me know if I should adjust the tests to ArgumentException instead of ArgumentNullException, or whether to put this back.

Reply:
@xoofx This one is left unanswered and the reason for the failing tests. Is it OK to have an ArgumentException or do you want to have the ArgumentNullException back?

Reply:
I think it can be folded into the Throw as commented separately.